Sources/Sentry/SentryHttpTransport.m

```diff
@@ -281,24 +290,36 @@ - (void)sendAllCachedEnvelopes
         self.isSending = YES;
     }
 
-    SentryFileContents *envelopeFileContents = [self.fileManager getOldestEnvelope];
-    if (nil == envelopeFileContents) {
-        SENTRY_LOG_DEBUG(@"No envelopes left to send.");
-        [self finishedSending];
-        return;
-    }
+    SentryEnvelope *envelope;
+    NSString *envelopeFilePath;
+
+    @synchronized(self.notStoredEnvelopes) {
+        if (self.notStoredEnvelopes.count > 0) {
+            envelope = self.notStoredEnvelopes[0];
+            [self.notStoredEnvelopes removeObjectAtIndex:0];
```

Comment:
`m`: I think we should only limit the synchronized block to this code. Especially `deleteEnvelopeAndSendNext` shouldn't be in the synchronized block. Something like this could do the trick:

```
@synchronized(self.notStoredEnvelopes) {
    if (self.notStoredEnvelopes.count > 0) {
        envelope = self.notStoredEnvelopes[0];
        [self.notStoredEnvelopes removeObjectAtIndex:0];
    }
}

if (envelope == nil) {
...
```